ao-tools/ao-bitbang/ao-bitbang.1
.\"
.\" Copyright © 2009 Keith Packard <keithp@keithp.com>
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful, but
.\" WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
.\" General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License along
.\" with this program; if not, write to the Free Software Foundation, Inc.,
.\" 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
.\"
.\"
.TH AO-BITBANG 1 "ao-bitbang" ""
.SH NAME
ao-bitbang \- Low-level cc1111 interface diagnostic tool
.SH SYNOPSIS
.B "ao-bitbang"
.SH DESCRIPTION
.I ao-bitbang
connects through a CP2103 usb-to-serial converter and uses the GPIO
pins to communicate with the debug port on a cc1111 device. It
provides raw access to the debug pins to help debug the lowest level
communication path.
.SH USAGE
.I ao-bitbang
reads a sequence of bit manipulations from stdin, sends them to the
device and reports status on stdout.
.P
Each line on stdin should contain a single letter command for each of
the three debug lines on the cc1111 -- clock, data and reset. Each bit
can be in one of three states -- on (C, D or R), off (.) or tri-state
(-) for input. Empty lines, or lines starting with '#' are
ignored. Anything beyond the last bit in a line is also ignored. The
bits must be listed in the correct order, and the 'on' values must
match the desired bit.
.SH EXAMPLE
.IP "Reset the target device"
.nf
 # reset
 C D R
 C D R
 C D R
 C D R
.fi
.IP "Get Chip ID"
.nf
 #
 # Debug mode - drive RESET_N low for two clock cycles
 #
 C D R
 . D .
 C D .
 . D .
 C D .
 . D R

 #
 # GET_CHIP_ID

 C . R	0
 . . R
 C D R	1
 . D R
 C D R	1
 . D R
 C . R	0
 . . R

 C D R	1
 . D R
 C . R	0
 . . R
 C . R	0
 . . R
 C . R	0
 . . R

 #
 # start reading again
 #

 C D R
 . - R
 C - R
 . - R
 C - R
 . - R
 C - R
 . - R

 C - R
 . - R
 C - R
 . - R
 C - R
 . - R
 C - R
 . - R

 C - R
 . - R
 C - R
 . - R
 C - R
 . - R
 C - R
 . - R

 C - R
 . - R
 C - R
 . - R
 C - R
 . - R
 C - R
 . - R

 C D R
.fi
.SH AUTHOR
Keith Packard


ChangeLog
commit 42ab6d52540d0326ef89e9d57954b08248558468
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 02:21:06 2009 -0600

    fix location for delivery of sources.list fragment

commit 977f5dc0bc7c666dcc1f21db77416efca0d696aa
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 02:11:11 2009 -0600

    update changelogs for Debian build

commit 8d4d6655f1b4c2fbc522fd255bfb75406e5ddaef
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 02:11:00 2009 -0600

    oops, aoview is now ao-view!

commit b9a97aea65f871fd287bc0bb566d8664766f4afd
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 02:07:16 2009 -0600

    update changelogs for Debian build

commit 21e3dd0affac89919e5d0e29c6e9eb1eacb51801
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 02:06:49 2009 -0600

    enable support for Debian menus

commit 0087c1776e0253fc2bd3b86f15bf9d1b32bdc45a
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 00:52:57 2009 -0600

    update changelogs for Debian build

commit 4486d9156e19e4280b42bcd422d81d04f2d04a92
Merge: dd09f0b 33edd62
Author: Bdale Garbee <bdale@gag.com>
Date:   Wed Aug 19 00:49:24 2009 -0600

    Merge branch 'master' of ssh://git.gag.com/scm/git/fw/altos

commit 33edd62992a32b0ec8ca66d879fa300871db5937
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 23:38:16 2009 -0700

    Update ao-view to add GPS satellite tracking data
    
    This adds another column to the display to hold per-satellite GPS
    tracking data and a count of the visible and locked sats.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 29687cbd462a332d9a36ed87500c5b737dcae3f4
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 22:35:15 2009 -0700

    Handle GPS satellite tracking data
    
    SiRF message #4 includes signal strength and GPS engine state for each
    of the satellites being tracked. This data is now parsed and sent to
    eeprom and the radio.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit dd09f0bc2b950c00f3b489878cd69ad8a003f46c
Author: Bdale Garbee <bdale@gag.com>
Date:   Tue Aug 18 21:57:01 2009 -0600

    update changelogs for Debian build

commit cd5ce661e2a8f9694933358ccb5b916fbed089c2
Author: Bdale Garbee <bdale@gag.com>
Date:   Tue Aug 18 21:49:39 2009 -0600

    add support for building Debian package

commit d996aa9b32fb0eb385bd3d158256c29788a42fe3
Merge: b3b2d3c 7d4ceb7
Author: Bdale Garbee <bdale@gag.com>
Date:   Tue Aug 18 18:56:09 2009 -0600

    Merge branch 'master' of ssh://git.gag.com/scm/git/fw/altos

commit 7d4ceb75a454e6c9b3fe0bd934fadcb5104dea36
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 17:55:22 2009 -0700

    Add ao-ejection.1 man page
    
    Document the input requirements and output format
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit b3b2d3c475a135084b5628c730fc6fca1ba0817b
Merge: 4685fc5 da12b89
Author: Bdale Garbee <bdale@gag.com>
Date:   Tue Aug 18 18:36:03 2009 -0600

    Merge branch 'master' of ssh://git.gag.com/scm/git/fw/altos

commit da12b89fb056a68e65ba363fef91d266727cb685
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 17:30:43 2009 -0700

    Create ChangeLog from git log

commit 7d69e2b3715faed10ce21ad562fc4d25dfc5f9c1
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 17:29:29 2009 -0700

    Fix ao-bitbang examples to not have . in the first column

commit 4685fc541466afbeefc151bcb64cd054739c048b
Merge: 1c2a0b6 c29275b
Author: Bdale Garbee <bdale@gag.com>
Date:   Tue Aug 18 18:09:38 2009 -0600

    Merge branch 'master' of ssh://git.gag.com/scm/git/fw/altos
    
    Conflicts:
    	ao-tools/ao-bitbang/Makefile.am
    	ao-tools/ao-eeprom/Makefile.am
    	ao-tools/ao-load/Makefile.am
    	ao-tools/ao-load/ao-load.c
    	ao-tools/ao-rawload/Makefile.am

commit c29275b72438637d46d7a50742882d2736eb176a
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 15:21:57 2009 -0700

    Add manual pages for remaining commands.
    
    Manuals written for ao-bitbang, ao-eeprom, ao-load, ao-rawload and
    ao-view.
    
    Manual for ao-dbg updated to reflect program name change.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 9b03d620722dc54630539afba40720c30de69b2d
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 12:19:31 2009 -0700

    Use --tty/-T on command line to specify target device
    
    Also, use the ALTOS_TTY environment variable in all tools. Note that
    the magic value of "BITBANG" switches the library to connecting
    through a CP2103 instead.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 7c790fe859dff062692964338091ffbbcdf63257
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 12:40:24 2009 -0700

    Rename tools to ao-<foo>
    
    Use a consistent prefix to make it easier to remember which programs
    belong to this package
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit 1c2a0b6653623b689d68d7349a6b2dce3e20a4a6
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 13:36:54 2009 -0700

    re-add debugger sources

commit 9a9cce5510b87252f863239ac807b9fb4395b288
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 12:19:31 2009 -0700

    Start working on using getopt for the tty name

commit 9789ca5e8caa9a013e804f307b9da380e147bd75
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 12:40:24 2009 -0700

    Rename tools to ao-<foo>
    
    Use a consistent prefix to make it easier to remember which programs
    belong to this package
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

commit a5782398d968e7cb11f7203afada7c216f233b3b
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 18 11:34:28 2009 -0700

    Remove unused cctools code paths for old libusb and cp2103 ioctls.
    
    Communication with the CP2103 board has gone through three revisions,
    first using ioctls supported by the CP2103 kernel driver, then using

ao-tools/ao-bitbang/ao-bitbang.c
/*
 * Copyright © 2009 Keith Packard <keithp@keithp.com>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
 */

#include "ccdbg.h"

int
main (int argc, char **argv)
{
	struct ccdbg	*dbg;

	dbg = ccdbg_open("BITBANG");
	if (!dbg)
		exit (1);

	ccdbg_add_debug(CC_DEBUG_BITBANG);

	ccdbg_manual(dbg, stdin);
}
